test/e2e-test-application/src/app/project/project.component.ts

```diff
@@ -128,6 +128,25 @@ export class ProjectComponent implements OnInit, OnDestroy {
     }
   }
 
+  openAsModal() {
+    linkManager().openAsModal('/projects/pr2/settings', {
+      title: 'microfrontend in a modal',
+      size: 'm',
+    }).then(() => {
+      const newListEl = document.createElement('li');
+      newListEl.textContent = 'promise resolved!';
+      newListEl.id = "promiseResolved";
+      newListEl.className = 'fd-list__item';
+      document.getElementById('promiseTest').appendChild(newListEl);
+    }).catch(() => {
+      const newListEl = document.createElement('li');
+      newListEl.textContent = 'promise rejected';
+      newListEl.id = "promiseRejected";
+      newListEl.className = 'fd-list__item';
+      document.getElementById('promiseTest').appendChild(newListEl);
+    });
+  }
```

Comment:
I can't think of any ways how/or when this would reject. I would omit the catch clause altogether. the then clause should be enough 👍

Reply:
I added this just in case there is a reject. I'll remove it👍